app/Http/Controllers/WaitingRoomsController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Gladiator\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Gladiator\Models\WaitingRoom;
+use Gladiator\Http\Controllers\Controller;
+
+class WaitingRoomsController extends Controller
+{
+    private $validationRules = [
+        'campaign_id' => 'required|numeric',
+        'campaign_run_id' => 'required|numeric',
+        'signup_start_date' => 'required|date',
+        'signup_end_date' => 'required|date',
+    ];
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $rooms = WaitingRoom::all();
+
+        return view('waitingrooms.index')->withRooms($rooms);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('waitingrooms.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, $this->validationRules);
+
+        $input = $request->all();
+
+        WaitingRoom::create($input);
+
+        $request->session()->flash('status', 'Waiting room has been saved!');
+
+        return redirect()->back();
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
```

Comment:
If you type hint a `WaitingRoom $room` in the parameters instead of `$id`, Laravel will automatically inject the model with a matching ID when routing to the controller! It's v slick.